zebra-network/src/protocol/external/codec.rs

```diff
@@ -674,6 +676,30 @@ impl Codec {
     fn read_filterclear<R: Read>(&self, mut _reader: R) -> Result<Message, Error> {
         Ok(Message::FilterClear)
     }
+
+    /// TBA
+    #[allow(clippy::unwrap_in_result)]
+    fn deserialize_transaction_spawning<R: Read + std::marker::Send>(
+        reader: R,
+    ) -> Result<Transaction, Error> {
+        let mut result = None;
+
+        // Correctness: TBA
+        //
+        // Since we use `block_in_place()`, other futures running on the connection task will be blocked:
+        // https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html
+        //
+        // We can't use `spawn_blocking()` because:
+        // - The `reader` has a lifetime (but we could replace it with a `Vec` of message data)
+        // - There is no way to check the blocking task's future for panics
```

Comment:
It seems like we have to use `tokio::task::block_in_place()` here - I added a comment with some detailed reasons.